service/src/java/org/apache/hive/service/cli/operation/HiveCommandOperation.java

```diff
@@ -124,14 +139,45 @@ public void runInternal() throws HiveSQLException {
       }
     } catch (CommandProcessorException e) {
       setState(OperationState.ERROR);
-      throw toSQLException("Error while processing statement", e);
+      throw toSQLException("Error while processing statement: " + e.toString(), e);
     } catch (Exception e) {
       setState(OperationState.ERROR);
       throw new HiveSQLException("Error running query: " + e.toString(), e);
     }
     setState(OperationState.FINISHED);
   }
 
+  private List<ProcessListInfo> getLiveQueryInfos(HiveSession parentSession) {
+    SessionManager sessionManager = parentSession.getSessionManager();
+    if (sessionManager == null) {
+      return null;
+    }
+    long currentTime = System.currentTimeMillis();
+    Collection<Operation> operations = sessionManager.getOperations();
+    return operations.stream()
+        .filter(op -> op instanceof SQLOperation) // Filter for SQLOperation instances
+        .map(op -> {
+          HiveSession session = op.getParentSession();
+          QueryInfo query = sessionManager.getOperationManager()
+              .getQueryInfo(op.getHandle().getHandleIdentifier().toString());
+
+          return new ProcessListInfo.Builder()
+              .setUserName(session.getUserName())
+              .setIpAddr(session.getIpAddress())
+              .setSessionId(session.getSessionHandle().getHandleIdentifier().toString())
+              .setSessionActiveTime((currentTime - session.getCreationTime()) / 1000)
+              .setSessionIdleTime((currentTime - session.getLastAccessTime()) / 1000)
+              .setQueryId(op.getQueryId())
+              .setExecutionEngine(query.getExecutionEngine())
+              .setBeginTime(query.getBeginTime())
```

Comment:
Nit: Can we make this timestamp human readable?

Reply:
We log it as is  in Hiverserver2 and also in web UI. But I think it makes sense to make it human readable here.